packages/components/autocomplete/src/use-autocomplete.ts

```diff
@@ -325,12 +334,17 @@ export function useAutocomplete<T extends object>(originalProps: UseAutocomplete
     }
   }, [isOpen]);
 
-  // unfocus the input when the popover closes & there's no selected item & no allows custom value
+  // react aria has different focus strategies internally
+  // hence, handle focus behaviours on our side for better flexibilty
+
   useEffect(() => {
-    if (!isOpen && !state.selectedItem && inputRef.current && !allowsCustomValue) {
-      inputRef.current.blur();
+    if (shouldFocus || isOpen) {
+      inputRef?.current?.focus();
+    } else {
+      inputRef?.current?.blur();
+      if (shouldFocus) setShouldFocus(false);
```

Comment:
@wingkwong make sure this doesn't cause infinite updates b/c `shouldFocus` is part of the `useEffect`  deps array and it is being updated inside the same `useEffect`